File: src/prefix-sum/2389.js
/**
 * @param {number[]} nums
 * @param {number[]} queries
 * @return {number[]}
 */
var answerQueries = function (nums, queries) {
  nums.sort((a, b) => {
    return a < b ? -1 : 1;
  });

  const prefixSums = [0];
  nums.forEach((num, index) => {
    prefixSums.push(prefixSums[index] + num);
  });

  return queries.map((query) => {
    let i = 1;
    for (; i < prefixSums.length; i++) {
      if (prefixSums[i] > query) {
        break;
      }
    }
    return i - 1;
  });
};

console.log(answerQueries([4, 5, 2, 1], [3, 10, 21]));
console.log(answerQueries([2, 3, 4, 5], [1, 2]));

File: src/prefix-sum/2389.cpp
class Solution
{
public:
  vector<int> answerQueries(vector<int> &nums, vector<int> &queries)
  {
    vector<int> prefixSums, result;

    sort(nums.begin(), nums.end());
    prefixSums.push_back(0);

    for (int i = 0; i < nums.size(); i++)
    {
      prefixSums.push_back(prefixSums[i] + nums[i]);
    }
    for (int i = 0; i < queries.size(); i++)
    {
      int j = 1;
      for (; j < prefixSums.size(); j++)
      {
        if (prefixSums[j] > queries[i])
        {
          break;
        }
      }
      result.push_back(j - 1);
    }

    return result;
  }
};